src/app.js

```diff
@@ -224,7 +209,15 @@ var run = function () {
     })
   })
   files.event.register('fileRemoved', function (path) {
-    if (path === ui.get('currentFile')) ui.set('currentFile', '')
+    editor.discard(path)
```

Comment:
I think `discard` must be after `switchToNextFile` otherwise the editor cannot discard it.